crates/bevy_egui/src/egui_ui.rs

```diff
@@ -0,0 +1,38 @@
+use bevy::prelude::*;
+
+use tracing::trace;
+
+pub struct EguiUi {
+    pub ui: Option<egui::Ui>,
+}
+
+impl FromResources for EguiUi {
+    fn from_resources(_resources: &Resources) -> Self {
+        Self {
+            ui: None,
+        }
+    }
+}
+
+pub(crate) struct EguiFrameStartEvent {
+    pub(crate) new_ui: egui::Ui,
+}
+
+// TODO(#56): move egui context updates to the start of the frame along with utilizing a context map, to remove the singleton restriction
+pub(crate) fn egui_state_update(
+    mut frame_start_events: ResMut<Events<EguiFrameStartEvent>>,
+    mut egui: ResMut<EguiUi>,
+) {
+    // Grab the newest ui provided by an event
+    let new_ui = frame_start_events.drain().next();
+
+    let EguiFrameStartEvent { new_ui } = if egui.ui.is_none() && new_ui.is_none() {
+        trace!("Did not find a UI this tick!");
+        return;
+    } else {
+        new_ui.expect("Previously an egui ui has been recieved, but one hasn't been recieved this frame. This can't happen!")
```

Comment:
```suggestion
        // Supposed to be infallible.
        new_ui.expect("Previously an egui UI had been received, but one hasn't been received this frame.")
```
Some grammatical and spelling errors, as well as moving the infallibility text to a comment. It's not helpful to put it in the actual message, and rather just takes space: if it can actually happen, it will, and if it can't, no reason to mark it as such as nobody will see it. Better to keep the message terse.